sdk/src/main/java/io/radar/sdk/Radar.kt

```diff
@@ -2225,6 +2241,91 @@ object Radar {
         })
     }
 
+    /**
+     * Sends a custom event.
+     *
+     * @see [](https://radar.io/documentation/api#sen-event)
+     *
+     * @param[name] The name of the event.
+     * @param[metadata] The metadata associated with the event.
+     * @param[callback] A callback.
+     */
+    @JvmStatic
+    fun sendEvent(name: String, metadata: JSONObject?, callback: RadarSendEventCallback) {
+        if (!initialized) {
+            callback.onComplete(RadarStatus.ERROR_PUBLISHABLE_KEY)
+
+            return
+        }
+
+        trackOnce(object : RadarTrackCallback {
+            override fun onComplete(status: RadarStatus, location: Location?, events: Array<RadarEvent>?, user: RadarUser?) {
+                if (status != RadarStatus.SUCCESS || location == null) {
+                    handler.post {
+                        callback.onComplete(status)
+                    }
+
+                    return
+                }
+
+                apiClient.sendEvent(name, metadata, user, object : RadarApiClient.RadarSendEventApiCallback {
+                    override fun onComplete(status: RadarStatus, res: JSONObject?, event: RadarEvent?) {
+                        handler.post {
+                            callback.onComplete(status, event)
+                        }
+                    }
+                })
+            }
+        })
+    }
+
+    /**
+     * Sends a custom event.
+     *
+     * @see [](https//radar.io/documentation/api#send-event)
+     *
+     * @param[name] The name of the event.
+     * @param[metadata] The metadata associated with the event.
+     * @param[block] A block callback
+     */
+    fun sendEvent(name: String, metadata: JSONObject?, block: (status: RadarStatus, event: RadarEvent?) -> Unit) {
+        sendEvent(
+            name,
+            metadata,
+            object : RadarSendEventCallback {
+                override fun onComplete(status: RadarStatus, event: RadarEvent?) {
+                    block(status, event)
+                }
+            }
+        )
+    }
+
+    /**
+     * Sends a custom event with a manually provided location.
+     *
+     * @see [](https://radar.io/documentation/api#send-event)
+     *
+     * @param[name] The name of the event.
+     * @param[location] The location of the event.
+     * @param[metadata] The metadata associated with the event.
+     * @param[callback] A callback.
+     *
+     * TODO (jsani)
+     */
+
+    /**
+     * Sends a custom event with a manually provided location.
+     *
+     * @see [](https://radar.io/documentation/api#send-event)
+     *
+     * @param[name] The name of the event.
+     * @param[location] The location of the event.
+     * @param[metadata] The metadata associated with the event.
+     * @param[block] A block callback.
+     *
+     * TODO (jsani)
+     */
+
```

Comment:
Should we leave these out (or at least not commit them) until they're implemented?